app/assets/javascripts/admin/index_utils/services/paged_fetcher.js.coffee

```diff
@@ -9,8 +9,9 @@ angular.module("admin.indexUtils").factory "PagedFetcher", (dataFetcher) ->
 
         if data.pages > 1
           for page in [2..data.pages]
-            dataFetcher(@urlForPage(url, page)).then (data) ->
+            lastPagePromise = dataFetcher(@urlForPage(url, page)).then (data) ->
               processData data
+          return lastPagePromise
```

Comment:
I just realised that this synchronicity may be the original objective of the code but coffee script confused the author. Maybe... 
The original code in javascript actually returns an _array_ of "results" which is an array of promises. The new code returns just the last promise and that, according to my tests, makes the calls synchronous.
My explanation for this is that an array of promises is taken to be a value that resolves the original promise (and not taken as being a promise), while returning a promise does not resolve the original promise until this returned promise is resolved itself (the last page).

The js code generated with the original code:

      PagedFetcher.prototype.fetch = function(url, processData) {
        return dataFetcher(this.urlForPage(url, 1)).then((function(_this) {
          return function(data) {
            var i, page, ref, results;
            processData(data);
            if (data.pages > 1) {
              results = [];
              for (page = i = 2, ref = data.pages; 2 <= ref ? i <= ref : i >= ref; page = 2 <= ref ? ++i : --i) {
                results.push(dataFetcher(_this.urlForPage(url, page)).then(function(data) {
                  return processData(data);
                }));
              }
              return results;
            }
          };
        })(this));
      };

and the new code :-)


PagedFetcher.prototype.fetch = function(url, processData) {
        return dataFetcher(this.urlForPage(url, 1)).then((function(_this) {
          return function(data) {
            var i, lastPagePromise, page, ref;
            processData(data);
            if (data.pages > 1) {
              for (page = i = 2, ref = data.pages; 2 <= ref ? i <= ref : i >= ref; page = 2 <= ref ? ++i : --i) {
                lastPagePromise = dataFetcher(_this.urlForPage(url, page)).then(function(data) {
                  return processData(data);
                });
              }
              return lastPagePromise;
            }
          };
        })(this));
      };